gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/impl/internal/mfa/filter/MfaSkipUserStronglyAuthFilter.java

```diff
@@ -37,15 +37,18 @@ public Boolean get() {
         // We need to check whether the AMFA, Device and Step Up rule is false since we don't know of other MFA return False
         final boolean mfaSkipped = context.isMfaSkipped();
         if (    // Whether Adaptive MFA is not true
-                context.isAmfaActive() && !context.isAmfaRuleTrue() ||
+                !mfaSkipped && context.isAmfaActive() && !context.isAmfaRuleTrue() ||
```

Comment:
Even if we know operator priority, I would prefer wrap the boolean operations between parenthesis for better readability. wdyt ?

Reply:
I don't mind doing this !
So it would be 
```suggestion
                !mfaSkipped && (context.isAmfaActive() && !context.isAmfaRuleTrue()) ||
```

?